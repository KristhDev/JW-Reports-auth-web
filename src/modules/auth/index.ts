export * from './components';
export * from './hooks';
export * from './layouts';
export * from './pages';
export * from './routes';