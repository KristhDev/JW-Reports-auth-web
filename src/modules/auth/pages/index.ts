export { Confirm } from './Confirm';
export { ResetPassword } from './ResetPassword';