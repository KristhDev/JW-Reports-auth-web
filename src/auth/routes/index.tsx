import { RouteObject } from 'react-router-dom';

import { Confirm, ResetPassword } from '../pages';

export const routes: RouteObject[] = [
    {
        path: '',
        element: <Confirm />
    },
    {
        path: 'reset-password',
        element: <ResetPassword />
    }
];