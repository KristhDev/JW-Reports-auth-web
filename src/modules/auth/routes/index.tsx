import { RouteObject } from 'react-router-dom';

/* Pages */
import { Confirm, ResetPassword } from '../pages';

/* Creating a route object. */
export const routes: RouteObject[] = [
    {
        path: '',
        element: <Confirm />
    },
    {
        path: 'reset-password',
        element: <ResetPassword />
    }
];