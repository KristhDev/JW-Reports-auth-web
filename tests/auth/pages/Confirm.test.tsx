import { render } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';

import { Confirm } from '../../../src/auth/pages';

const confirmRouter = createMemoryRouter([
    {
        path: '',
        element: <Confirm />
    }
], { initialEntries: [ '#message=Test+message' ] });

const confirmSignUpRouter = createMemoryRouter([
    {
        path: '',
        element: <Confirm />
    }
], { initialEntries: [ '#type=signup' ] });

describe('Test in <Confirm /> page', () => {
    it('should to match the snapshot', () => {
        const { container } = render(<RouterProvider router={ confirmRouter } />);
        expect(container).toMatchSnapshot();
    });

    it('should render the message', () => {
        const { getByRole } = render(<RouterProvider router={ confirmRouter } />);
        const msgText = getByRole('heading', { level: 2 });

        expect(msgText).toBeTruthy();
        expect(msgText.innerHTML).toBe('Test message');
    });

    it('should render sign up message', () => {
        const { getByRole } = render(<RouterProvider router={ confirmSignUpRouter } />);
        const msgText = getByRole('heading', { level: 2 });

        expect(msgText).toBeTruthy();
        expect(msgText.innerHTML).toBe('Haz verificado tu cuenta correctamente, abre la aplicaci√≥n e ingresa con tu cuenta.');
    });
});