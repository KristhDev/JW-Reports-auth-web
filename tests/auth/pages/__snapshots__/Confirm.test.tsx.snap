// Vitest Snapshot v1

exports[`Test in <Confirm /> page > should to match the snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title-container undefined"
      data-testid="title-container"
    >
      <h1
        class="title-text undefined"
      >
        Confirmaci√≥n exitosa
      </h1>
    </div>
    <div
      class="content"
    >
      <h2>
        Test message
      </h2>
    </div>
  </div>
</div>
`;
